mergekit/architecture.py

```diff
@@ -122,29 +123,82 @@ def has_defined_spaces(self) -> bool:
         """
         return False
 
+    @abstractmethod
+    def _substitute(
+        self,
+        item: Union[WeightInfo, ProceduralSpaceInfo],
+        config: PretrainedConfig,
+        layer_idx: Optional[int] = None,
+    ) -> Union[WeightInfo, ProceduralSpaceInfo]:
+        """Substitute any template variables in the item with values from the config"""
+        ...
+
+
+class MappingInfo(BaseModel, frozen=True):
+    """Information about a mapping between two models.
+
+    Attributes:
+        from_model (str):
+            The name of the model from which the mapping originates.
+        to_model (str):
+            The name of the model to which the mapping applies.
+    """
+
+    start_architectures: List[str]
+    destination_architectures: List[str]
+    weights_mapping: Dict[str, str]
+
 
 class ConfiguredArchitectureInfo(BaseModel, frozen=True, arbitrary_types_allowed=True):
     info: ArchitectureInfo
     config: PretrainedConfig
+    overrides: Optional[Dict[str, str]] = None
+
+    def _substitute_name(self, weight: WeightInfo) -> WeightInfo:
+        if self.overrides and weight.name in self.overrides:
+            return self.info.WeightInfo(
+                name=self.overrides[weight.name], **self.model_dump(exclude=["name"])
+            )
+        return weight
 
     def num_layers(self) -> int:
         return self.info.num_layers(self.config)
 
     def pre_weights(self) -> List[WeightInfo]:
-        return self.info.pre_weights(self.config)
+        return [
+            self._substitute_name(wi)
+            for wi in self.info.pre_weights(self.config, self.overrides.pre_weights)
+        ]
 
     def post_weights(self) -> List[WeightInfo]:
-        return self.info.post_weights(self.config)
+        return [
+            self._substitute_name(wi)
+            for wi in self.info.post_weights(self.config, self.overrides.pre_weights)
+        ]
 
     def layer_weights(self, index: int) -> List[WeightInfo]:
-        return self.info.layer_weights(index, self.config)
+        return [
+            self._substitute_name(wi)
+            for wi in self.info.layer_weights(index, self.config)
+        ]
 
     def procedural_spaces(self) -> List[ProceduralSpaceInfo]:
         return self.info.procedural_spaces(self.config)
 
     def all_weights(self) -> List[WeightInfo]:
         return self.info.all_weights(self.config)
 
+    def set_overrides(self, overrides: Dict[str, str]) -> "ConfiguredArchitectureInfo":
+        # NOTE: this makes sure template strings in overrides are filled in
+        overrides = {
+            self.info._substitute(k, self.config): self.info._substitute(v, self.config)
```

Comment:
:+1: I think we actually need to loop over layers here though - if there's a `${layer_index}` reference in a template we need to populate it for each value of `layer_index`. (If you have thoughts on how to make the whole template setup cleaner I'd definitely appreciate them - it's a little silly at the moment.)

Reply:
I was careless. I set out out to do exactly that and somehow meandered away. I will make this correction